src/textdynamics.ts

```diff
@@ -91,6 +91,10 @@ export class TextDynamics extends Note {
   /** Preformat the dynamics text. */
   preFormat(): this {
     let total_width = 0;
+    // Issue 1122: length of this.glyphs must be <=
```

Comment:
You can remove this issue number.

Note that the standard pattern we use in other classes is to define a `this.preFormatted` boolean in the constructor.

Then here you just `if (preFormatted) return;`

I'd prefer we do that to stay consistent unless there's a good reason to do something different.

Reply:
@0xfe  that is what I thought at first.  But looking into it, this is actually the same behavior as `StaveNote`.  When `formatter.format` is called, it calls `createTickContext` which sets preformat back to false.  So it is consistent, but not optimal.

The larger can of worms is that we should keep formatter from creating the tick contexts a second time.  But I think we might also need to add some checks to make sure that no modifiers etc. were added between the calls to `preCalculateMinTotalWidth` and `format`' that might affect the width calculation.  And I wasn't sure if there were other ways the list of tick contexts could be non-empty.  This would result in some tickables/modifiers being preformatted, and others not.

I'm happy to take this on also, but I don't know if it should be for this PR.  I know that the change for TextDynamics won't be a breaking change, it's harder to be sure with Formatter since it is used in different ways.